Add folder_include_patterns and file_include_patterns

ferhtgoldaraz opened the pull request:
Extend the documentation of project folder declaration explaining a bit further it's optional values, and adding two that were not being mentioned

Please note, english is not my first language. It would be appreciated if you could read and ammend the changes if they happen to sound weird or something.

ferhtgoldaraz committed:
Add folder_include_patterns and file_include_patterns

Extend the documentation of project folder declaration explaining a bit further it's optional values, and adding two that were not being mentioned

Please note, english is not my first language. It would be appreciated if you could read and ammend the changes if they happen to sound weird or something.

FichteFoll commented on source/file_management/file_management.rst:
```diff
@@ -223,11 +223,52 @@ and ``build_systems``, for project-specific build systems.
+    and ``file_exclude_patterns``. They all admit an array of strings
+    as values, and understand basic globbing syntax, such as "*.html".
+    
+    The ones that include files and folders, work by excluding
```
You are referring to the "patterns" here, not the settings. While it can be interpreted to be the patterns, explicitly mentioning it is better. (It confused me at first.)
Same for the second part of this paragraph.

FichteFoll commented on source/file_management/file_management.rst:
```diff
@@ -223,11 +223,52 @@ and ``build_systems``, for project-specific build systems.
+    
+    Note that a pattern that includes any folder in the project
+    directory will not automatically include every of it's sub-folders,
+    but including theese will include any other descendants.
```
This paragraph does not make 100% sense to me. Mind explaining?

Also *these

Edit: It appears you are explaining it below but it doesn't completely get through. I think we need to reword this.

FichteFoll commented on source/file_management/file_management.rst:
```diff
@@ -223,11 +223,52 @@ and ``build_systems``, for project-specific build systems.
+    though with lesser precedence.
+    
+    Note that a pattern that includes any folder in the project
+    directory will not automatically include every of it's sub-folders,
```
*its

FichteFoll commented on source/file_management/file_management.rst:
```diff
@@ -223,11 +223,52 @@ and ``build_systems``, for project-specific build systems.
+    "bar" inside the "folder "foo" that resides in the project folder
+    will not be included automatically, whereas another folder called
+    "bar" or any other name in some other "foo" folder nested deeper
+    **wll** be included by this definition. If you want to include
```
*will

FichteFoll commented on source/file_management/file_management.rst:
```diff
@@ -223,11 +223,52 @@ and ``build_systems``, for project-specific build systems.
+        }
+    
+    If the folder "foo" is included, that will by itself include
+    any  other folder whose name starts with foo, whether it is in
```
double spaces here

FichteFoll commented on source/file_management/file_management.rst:
```diff
@@ -223,11 +223,52 @@ and ``build_systems``, for project-specific build systems.
+    If the folder "foo" is included, that will by itself include
+    any  other folder whose name starts with foo, whether it is in
+    the project folder, or inside some other folder. But, the folder
+    "bar" inside the "folder "foo" that resides in the project folder
```
-"

So, do you mean a folder `foo/bar` here? Writing that would be easier to understand than the long sentence.

Do it like so: `:file:`foo/bar``

ferhtgoldaraz replied:
I'm not being able to make `:file:`foo/bar`` work I'm afraid

FichteFoll commented on source/file_management/file_management.rst:
```diff
@@ -223,11 +223,52 @@ and ``build_systems``, for project-specific build systems.
+    "bar" or any other name in some other "foo" folder nested deeper
+    **wll** be included by this definition. If you want to include
+    every sub-folder form a folder residing in the project folder,
+    you can use something like "["foo", foo/*]".
```
Use literal blocks for this: ```["foo", "foo/*"]```

FichteFoll commented on source/file_management/file_management.rst:
```diff
@@ -223,11 +223,52 @@ and ``build_systems``, for project-specific build systems.
+    
+    Note also that a rule like the one in the prevous example would
+    not let us exclude an already included "foo" sub-folder through
+    the use of "folder_exclude_patterns", no matter the specificity
```
Here and below, please make use of literal blocks, so ``folder_exclude_patterns`` and `:file:`baz/bar/foo``.

FichteFoll commented:
Thanks for your contribution, I added a few comments.

Please try to restructure your additions to use semantic linefeeds as mentioned [here](https://github.com/guillermooo/sublime-undocs/blob/sublime-text-3/CONTRIBUTING.md#line-widths).

ferhtgoldaraz committed:
Edited project file and folder pattern rules explanation

As per some comments made, I've tried my best to make the explacantion a bit clearer, and added the suggested edits.

ferhtgoldaraz committed:
Little syntax mistakes and syle editions

ferhtgoldaraz commented:
Thanks for your input, I've tried my best to improve the explanation and to update the style. I've done what I can with semantic line feeds, but I might have not got it, so feel free to reformat at will

ferhtgoldaraz committed:
Fixed the examples not being shown

ferhtgoldaraz committed:
Removed :file:, It doesn't seem to work

ferhtgoldaraz commented:
There are some file references written as ``foo`` that are not working. I'm gonna leave tham as `foo` for now, as `:file:`foo`` doesn't seem to work. I'm afraid I'm not familiar with that syntax

FichteFoll commented:
The file role is define in Sphinx [here](http://sphinx-doc.org/markup/inline.html#role-file). I'm not sure where you are seeing that the file role wouldn't work because it's being used throughout the code base (for example [here](http://docs.sublimetext.info/en/latest/basic_concepts.html#the-data-directory)).
Either way, just ``foo`` would be worse since restructured text requires double backticks for inline literals, that is ```foo```.

I will take a look at your changes in detail later.

Edit: It appears that it will probably easier if I just take whatever you have here (by squashing probably) and then do my own pass.

ferhtgoldaraz committed:
made folders inline code

ferhtgoldaraz committed:
fixed another typo

ferhtgoldaraz committed:
fixed one last folder reference
